modules/ingester/instance.go

```diff
@@ -66,6 +67,11 @@ type instance struct {
 	completingBlocks []*wal.AppendBlock
 	completeBlocks   []*wal.LocalBlock
 
+	searchHeadBlock      *searchStreamingBlockEntry
+	searchAppendBlocks   map[*wal.AppendBlock]*searchStreamingBlockEntry
```

Comment:
is it the long term plan to maintain a set of search blocks separate from existing tempo blocks or to pull this functionality together?

Reply:
TBD.  TempoDB is historically an agnostic KV store and integrating search would change that.  It was considered, but starting out as separate seems the best either way.